### PR TITLE
Stevenic/4.6 state **FOR REVIEW**

### PR DESCRIPTION
**Pull request by Stevenic**
## Description
Adds a new DialogStateManager class to DialogContext.state. This is needed for Adaptive Dialogs.

## Specific Changes
New DialogStateManager an related classes.

## Testing
Working on Unit Tests and Docs...

### CODE REVIEW

File: libraries/botbuilder-dialogs/src/state/pathResolver.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+ * @module botbuilder-dialogs
+ */
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { DialogContext } from "../dialogContext";
+
+export interface PathResolver {
+    matched(dc: DialogContext, path: string): boolean;
```
Comment (tomlm) [inline libraries/botbuilder-dialogs/src/state/pathResolver.ts]: >DialogContext [](start = 16, length = 13)

no need for dialogContext

File: libraries/botbuilder-dialogs/src/state/pathResolver.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+    getValue(dc: DialogContext, memory: object, path: string, defaultValue?: any): any;
```
Comment (tomlm) [inline libraries/botbuilder-dialogs/src/state/pathResolver.ts]: >c: DialogContext [](start = 14, length = 16)

no need for dialogContext

File: libraries/botbuilder-dialogs/src/state/pathResolver.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    setValue(dc: DialogContext, memory: object, path: string, value: any): void;
```
Comment (tomlm) [inline libraries/botbuilder-dialogs/src/state/pathResolver.ts]: >c: DialogContext [](start = 14, length = 16)

no need for dialogContext

File: libraries/botbuilder-dialogs/src/state/pathResolver.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+    removeValue(dc: DialogContext, memory: object, path: string): void;
```
Comment (tomlm) [inline libraries/botbuilder-dialogs/src/state/pathResolver.ts]: >dc: DialogContext [](start = 16, length = 17)

no need for dialogContext

**Comment by cleemullins** [comment]
Closing per discussion with @microsoft/intelligent-vending-machines-sme-group

**Pull request closed without merging**